### DISCUSSION TITLE
adding new row with a nullable unique field. Unexpectedly deleted the data in another row

### DISCUSSION THREAD

**Discussion by muhajirdev** [prisma/prisma#15635] [Q&A]
I have this model 

```
model User {
  id        String   @id @default(cuid())
  circleProfile     CircleProfile?
}

model CircleProfile {
  id              String    @id @default(cuid())
  user          User?     @relation(fields: [userId], references: [id], onDelete: SetNull)
  userId       String?   @unique
}
```

now, say my data looks like this
```
User
user1
user2

CircleProfile
profile1 -> userId: user1
profile2 -> userId: user2

```

And now execute this code

``` typescript
db.circleProfile.create({
    data: {
      id: 'profile3',
      user: {
        connect: { id: 'user1' },
      },
    },
  })
```

what will happen is, my data becomes like this

```
User
user1
user2

CircleProfile
profile1 -> userId: null
profile2 -> userId: user2
profile3 -> userId: userId1

```

By the way this also reproducible with prisma studio. All I need to do is just
- create `profile3` with empty userId`
- now update `profile3` `userId` with user1
- `profile1`'s `userId` will be null

This is kinda unexpected. I tried with normal postgresql client ( [postico](https://eggerapps.at/postico/) ). I throws error when I changed updated the userId

<img width="275" alt="image" src="https://user-images.githubusercontent.com/12745166/193416575-0495883f-8c1b-438a-8fbc-0680f0973190.png">

But in prisma, it doesn't throw the error? Am I missing something here? 
I took a look at referential actions. There seems nothing mentioning this case (adding new row with a unique field, where the data is already used)

**Comment by austincrim** [comment]
Hey @muhajirdev 👋,

I was able to recreate this as well, and it seems like unexpected behavior. Would you mind opening a [bug report](https://github.com/prisma/prisma/issues/new?assignees=&labels=kind%2Fbug&template=bug_report.yml) with all the relevant information?

**Reply by vladi-stevanovic** [reply]
Hello @muhajirdev did you perhaps end up creating that bug report? Let us know the link so our team can look further into this!

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
